src/auto/injector.js

```diff
@@ -100,7 +100,7 @@ function annotate(fn, strictDi, name) {
       last;
 
   if (typeof fn === 'function') {
-    if (!($inject = fn.$inject)) {
+    if (!($inject = fn.hasOwnProperty('$inject'))) {
```

Comment:
I assume you meant something like https://github.com/angular/angular.js/pull/15538/files#diff-ced481bc839279a343e9f8bb8dcf6e98R103.